I've just noticed there are errors in the log
```
11-12 17:16:12.899 22088-22088/org.wordpress.android E/WordPress-API: Volley error on https://public-api.wordpress.com/rest/v1.1/sites/30121039/stats/comments/?max=6&locale=en_US
    com.android.volley.ParseError: com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was BOOLEAN at line 1 column 880 path $.authors[1].follow_data
        at org.wordpress.android.fluxc.network.rest.GsonRequest.parseNetworkResponse(GsonRequest.java:89)
        at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:125)
     Caused by: com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was BOOLEAN at line 1 column 880 path $.authors[1].follow_data
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:226)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222)
        at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:41)
        at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:82)
        at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:61)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222)
        at com.google.gson.Gson.fromJson(Gson.java:927)
        at com.google.gson.Gson.fromJson(Gson.java:892)
        at com.google.gson.Gson.fromJson(Gson.java:841)
        at com.google.gson.Gson.fromJson(Gson.java:813)
        at org.wordpress.android.fluxc.network.rest.GsonRequest.parseNetworkResponse(GsonRequest.java:83)
        at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:125) 
     Caused by: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was BOOLEAN at line 1 column 880 path $.authors[1].follow_data
        at com.google.gson.stream.JsonReader.beginObject(JsonReader.java:385)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:215)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131) 
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222) 
        at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:41) 
        at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:82) 
        at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:61) 
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131) 
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222) 
        at com.google.gson.Gson.fromJson(Gson.java:927) 
        at com.google.gson.Gson.fromJson(Gson.java:892) 
        at com.google.gson.Gson.fromJson(Gson.java:841) 
        at com.google.gson.Gson.fromJson(Gson.java:813) 
        at org.wordpress.android.fluxc.network.rest.GsonRequest.parseNetworkResponse(GsonRequest.java:83) 
        at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:125) 
11-12 17:16:12.901 22088-22088/org.wordpress.android D/WordPress-API: Receiving OnUnexpectedError event, message: com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was BOOLEAN at line 1 column 880 path $.authors[1].follow_data
```

--- a/WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/StatsListFragment.kt
+++ b/WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/StatsListFragment.kt
@@ -26,8 +26,10 @@ import org.wordpress.android.ui.stats.refresh.NavigationTarget.ViewFollowersStat
 import org.wordpress.android.ui.stats.refresh.NavigationTarget.ViewPostDetailStats
 import org.wordpress.android.ui.stats.refresh.StatsListViewModel.StatsListType
 import org.wordpress.android.util.DisplayUtils
+import org.wordpress.android.util.Event
 import org.wordpress.android.util.ToastUtils
 import org.wordpress.android.util.image.ImageManager
+import org.wordpress.android.util.observeEvent
 import org.wordpress.android.widgets.RecyclerItemDecoration
 import javax.inject.Inject
 
@@ -111,7 +113,7 @@ class StatsListFragment : Fragment() {
             }
         })
 
-        viewModel.navigationTarget.observe(this, Observer {
+        viewModel.navigationTarget.observeEvent(this) {
             when (it) {
                 is AddNewPost -> ActivityLauncher.addNewPostForResult(activity, site, false)
                 is SharePost -> {
@@ -142,7 +144,8 @@ class StatsListFragment : Fragment() {
                     ActivityLauncher.viewCommentsStats(activity, site)
                 }
             }
-        })
+            true
+        }
     }
 
     private fun updateInsights(insightsState: InsightsUiState) {
@@ -157,7 +160,7 @@ class StatsListFragment : Fragment() {
     }
 }
 
-sealed class NavigationTarget {
+sealed class NavigationTarget : Event() {
     object AddNewPost : NavigationTarget()
     data class SharePost(val url: String, val title: String) : NavigationTarget()
     data class ViewPostDetailStats(

--- a/WordPress/src/main/java/org/wordpress/android/util/LiveDataUtils.kt
+++ b/WordPress/src/main/java/org/wordpress/android/util/LiveDataUtils.kt
@@ -1,7 +1,9 @@
 package org.wordpress.android.util
 
+import android.arch.lifecycle.LifecycleOwner
 import android.arch.lifecycle.LiveData
 import android.arch.lifecycle.MediatorLiveData
+import android.arch.lifecycle.Observer
 
 fun <T> merge(vararg sources: LiveData<T>): LiveData<T> {
     val mediator = MediatorLiveData<T>()
@@ -12,3 +14,18 @@ fun <T> merge(vararg sources: LiveData<T>): LiveData<T> {
     }
     return mediator
 }
+
+/**
+ * Use this in order to observe an emission only once - for example for displaying a Snackbar message
+ */
+fun <T : Event> LiveData<T>.observeEvent(owner: LifecycleOwner, observer: (T) -> Boolean) {
+    this.observe(owner, Observer {
+        if (it != null && !it.hasBeenHandled) {
+            if (observer(it)) {
+                it.hasBeenHandled = true
+            }
+        }
+    })
+}
+
+open class Event(var hasBeenHandled: Boolean = false)

--- a/build.gradle
+++ b/build.gradle
@@ -69,5 +69,5 @@ subprojects {
 }
 
 ext {
-    fluxCVersion = 'f7c231a93f51c5a7cd52209687232bc351ae5e4e'
+    fluxCVersion = '3de2ac4233162226f014d736a349cb58d105e3aa'
 }

--- a/build.gradle
+++ b/build.gradle
@@ -69,5 +69,5 @@ subprojects {
 }
 
 ext {
-    fluxCVersion = 'eb943acfcdab8e383f279cac9a53992b4133067e'
+    fluxCVersion = 'f7c231a93f51c5a7cd52209687232bc351ae5e4e'
 }

